### DISCUSSION TITLE
What is the correct return type for mixins? (simple case)

### DISCUSSION THREAD

**Discussion by onadeemdev** [MichalLytek/type-graphql#1436] [Q&A]
Hi when we use mixins what is the correct return type for the function? I am trying to properly define it but could use some help, as somewhat new to TS. Does this look correct? 

```import { Field, InputType, ClassType } from 'type-graphql'

type EmailMixinReturnType<TClassType extends ClassType> = new (
  ...args: ConstructorParameters<TClassType>
) => InstanceType<TClassType> & { email: string }

const EmailMixin = <TClassType extends ClassType>(BaseClass: TClassType): EmailMixinReturnType<TClassType> => {
  @InputType('EmailInput')
  class EmailInput extends BaseClass {
    @Field()
    email: string
  }
  return EmailInput
}

export default EmailMixin

**Comment by MichalLytek** [comment]
Why do you want to define the type? You should rely on type inference.

**Comment by onadeemdev** [comment]
It's for curiosity, I am using a TS project that requires defining the
function return type. I can disable the line but would also just like to
know to help me improve with TS. Thanks!

**Reply by MichalLytek** [reply]
Honestly, I have no idea. This lint rules only make things worse for such cases like mixins.

**Reply by onadeemdev** [reply]
ah ok thanks anyway!
